packages/react-components/react-progress/src/components/Progress/useProgressStyles.ts

```diff
@@ -1,33 +1,200 @@
-import { makeStyles, mergeClasses } from '@griffel/react';
-import type { ProgressSlots, ProgressState } from './Progress.types';
+import { makeStyles, mergeClasses, shorthands } from '@griffel/react';
+import { tokens, typographyStyles } from '@fluentui/react-theme';
+import { useFluent_unstable as useFluent } from '@fluentui/react-shared-contexts';
+import type { ProgressState, ProgressSlots } from './Progress.types';
 import type { SlotClassNames } from '@fluentui/react-utilities';
 
 export const progressClassNames: SlotClassNames<ProgressSlots> = {
   root: 'fui-Progress',
-  // TODO: add class names for all slots on ProgressSlots.
-  // Should be of the form `<slotName>: 'fui-Progress__<slotName>`
+  bar: 'fui-Progress__bar',
+  track: 'fui-Progress__track',
+  label: 'fui-Progress__label',
+  description: 'fui-Progress__description',
+};
+
+// Internal CSS vars
+export const progressCssVars = {
+  percentageCssVar: '--fui-Progress--percentage',
+  transitionCssVar: '--fui-Progress--transition',
+};
+
+const barThicknessValues = {
+  default: '2px',
+  large: '4px',
+};
+
+const IndeterminateProgress = {
+  '0%': {
+    left: '0%',
+  },
+  '100%': {
+    left: '100%',
+  },
+};
+
+const IndeterminateProgressRTL = {
+  '0%': {
+    right: '-30%',
+  },
+  '100%': {
+    right: '100%',
+  },
 };
 
 /**
  * Styles for the root slot
  */
-const useStyles = makeStyles({
+const useRootStyles = makeStyles({
   root: {
-    // TODO Add default styles for the root element
+    display: 'grid',
+    ...shorthands.gap('8px', '0px'),
+    ...shorthands.overflow('hidden'),
   },
+});
 
-  // TODO add additional classes for different states and/or slots
+/**
+ * Styles for the title
+ */
+const useLabelStyles = makeStyles({
+  default: {
+    gridRowStart: '1',
+    ...typographyStyles.body1,
+    color: tokens.colorNeutralForeground1,
+  },
+});
+
+/**
+ * Styles for the description
+ */
+const useDescriptionStyles = makeStyles({
+  default: {
+    gridRowStart: '3',
+    ...typographyStyles.caption1,
+    color: tokens.colorNeutralForeground2,
+  },
+});
+
+/**
+ * Styles for the progress bar
+ */
+const useBarStyles = makeStyles({
+  base: {
+    gridColumnStart: '1',
+    gridRowStart: '2',
+    backgroundColor: tokens.colorCompoundBrandBackground,
+
+    '@media screen and (forced-colors: active)': {
+      backgroundColor: 'Highlight',
+    },
+  },
+  defaultHeight: {
+    height: barThicknessValues.default,
+  },
+  largeHeight: {
+    height: barThicknessValues.large,
+  },
+  determinate: {
+    width: `var(${progressCssVars.percentageCssVar})`,
+    transitionProperty: `var(${progressCssVars.transitionCssVar})`,
+  },
+  indeterminate: {
+    width: 0,
+    transitionProperty: 'width .3s ease',
```

Comment:
Presumably this needs to be separated out into transitionProperty, transitionDuration, and transitionTimingFunction, right?

That said, you aren't using transitions for indeterminate right? Just for determinate...